Create README.md for new integrations docs/structure

justinegeffen opened the pull request:
Moving the integrations system admin content from docs.mattermost.com to developers.mattermost.com so they're all in one place making it more logical, and easier, to locate information.

Still in progress:
- Moving images across.
- Converting tables.

~~Known issue: The build isn't working for some reason and I am looking into that.~~

Note:
This should be merged on 16 June (Wednesday). No content has been changed for this PR, just formatting. Next PR will separate the integrations section into system admin and developer, and content review will be done.

~~ETA: @hanzei, I've created an images directory as it's neater but the images only seem to work if they're in the admin-guide folder. I'm using ``` ![image](oauth2_deauthorize_app.png)``` as the format. Is there a way to get the images to display from the images folder within the admin guide?~~

cwarnermm commented on site/content/integrate/admin-guide/admin-zapier-integration.md:
```diff
@@ -0,0 +1,148 @@
+
+- **Markdown:** Mattermost supports the use of [Markdown](https://docs.mattermost.com/help/messaging/formatting-text.html) in Zapier integrations. For example, use [heading markdown](https://docs.mattermost.com/help/messaging/formatting-text.html#headings) for Jira issue titles.
+- **Custom Icons:** Use different icons for different services and Zapier integrations.
+- **Hashtags:** Use hashtags to make your Zapier posts searchable. Use different hashtags for different services and Zapier integrations. For example, use the dynamic fields available in Zapier to include ticket a Jira ticket number in hashtags. This makes all conversation on a specific ticket instantly searchable by clicking the hashtag.
```
```suggestion
- **Hashtags:** Use hashtags to make your Zapier posts searchable. Use different hashtags for different services and Zapier integrations. For example, use the dynamic fields available in Zapier to include ticket a Jira ticket number in hashtags. This makes all conversation on a specific ticket instantly searchable by selecting the hashtag.```

cwarnermm commented on site/content/integrate/admin-guide/admin-zapier-integration.md:
```diff
@@ -0,0 +1,148 @@
+
+### Deauthorize the Zapier App
+
+If you'd like to deauthorize Zapier so it can no longer post through your connected account, go to **Account Settings > Security > OAuth 2.0 Applications**, then click **Deauthorize** on the Zapier app.
```
```suggestion
If you'd like to deauthorize Zapier so it can no longer post through your connected account, go to **Account Settings > Security > OAuth 2.0 Applications**, then select **Deauthorize** on the Zapier app.
```

cwarnermm commented on site/content/integrate/admin-guide/admin-embedding.md:
```diff
@@ -0,0 +1,69 @@
+weight: 80
+---
+
+## Launching Mattermost From a Button Click
```
```suggestion
## Launching Mattermost From a Button Selection
```

cwarnermm commented on site/content/integrate/admin-guide/admin-personal-access-token.md:
```diff
@@ -0,0 +1,78 @@
+### System Console
+
+1. Go to **System Console > Users**, search for the user account which the token belongs to, then select **Manage Tokens** from the dropdown.
+2. Identify the access token you want to revoke and hit **Delete**. Confirm the deletion.
```
```suggestion
2. Identify the access token you want to revoke, then select **Delete** and confirm the deletion.
```

justinegeffen commented on site/content/integrate/admin-guide/admin-interactive-messages.md:
```diff
@@ -0,0 +1,349 @@
+
+## Share Your Integration
+
+If you've built an integration for Mattermost, please consider [sharing your work](https://mattermost.org/share-your-mattermost-projects) in our [app directory](https://integrations.mattermost.com).
```
```suggestion
If you've built an integration for Mattermost, please consider sharing your work in our [app directory](https://integrations.mattermost.com).
```

justinegeffen commented on site/content/integrate/admin-guide/admin-bot-accounts.md:
```diff
@@ -0,0 +1,185 @@
+
+### Do bot access tokens expire?
+
+No, but you can automate your integration to cycle its token [through the REST API](https://api.mattermost.com/#tag/users%2Fpaths%2F~1users~1%7Buser_id%7D~1tokens%2Fpost).
```
@hanzei, can you confirm whether https://api.mattermost.com/#operation/EnableUserAccessToken is the correct API call?

hanzei commented on site/content/integrate/admin-guide/admin-interactive-messages/_index.md:
````diff
@@ -0,0 +1,347 @@
+
+The following payload gives an example that uses message buttons.
+
+```
````
The indentation is 2 spaces off in the whole file.

hanzei commented on site/content/integrate/admin-guide/admin-message-attachments/_index.md:
````diff
@@ -0,0 +1,159 @@
+Here is an example message attachment:
+
+```
+  :linenos:
````
Copy past error

hanzei commented on site/content/integrate/admin-guide/admin-message-attachments/_index.md:
````diff
@@ -0,0 +1,159 @@
+```
+  :linenos:
+
+  {
````
The indentation is also 2 spaces off.

hanzei commented on site/content/integrate/admin-guide/admin-message-attachments/_index.md:
```diff
@@ -0,0 +1,159 @@
+---
```
Can we simplify the file name to `site/content/integrate/admin-guide/message-attachments/_index.md`

justinegeffen replied:
That's a good idea.

hanzei commented on site/content/integrate/admin-guide/admin-oauth2/_index.md:
```diff
@@ -0,0 +1,80 @@
+1. Log in to your Mattermost server as the System Administrator.
+2. Go to **System Console > Integrations > Integration Management**.
+3. Set [Enable OAuth 2.0 Service Provider](https://docs.mattermost.com/administration/config-settings.html#enable-oauth-2-0-service-provider) to **True**.
+4. (Optional) If you'd like to allow external applications to post with customizable usernames and profile pictures, then set [Enable integrations to override usernames](https://docs.mattermost.com/administration/config-settings.html#enable-webhooks-and-slash-commands-to-override-usernames) and [Enable integrations to override profile picture icons](https://docs.mattermost.com/administration/conf ...
```
Link is wrong.

hanzei commented on site/content/integrate/admin-guide/admin-personal-access-token/_index.md:
```diff
@@ -0,0 +1,75 @@
+2. Identify the account you want to create a personal access token with. You may optionally create a new user account for your integration, such as for a bot account. By default, only System Admins have permissions to create a personal access token.
+3. To create an access token with a non-admin account, you must first give it the appropriate permissions. Go to **System Console > User Management > Users**, search for the user account, then select **Manage Roles** from the dropdown.
+
+![image](images/access_token_manage_roles.png)
```
Images are broken

hanzei commented on site/content/integrate/admin-guide/admin-personal-access-token/_index.md:
```diff
@@ -0,0 +1,75 @@
+---
+title: "Personal Access Tokens"
+heading: "Introduction"
+description: "Personal Access Tokens for System Admins."
```
It's not only for System admins.

hanzei commented on site/content/integrate/admin-guide/admin-slash-commands/_index.md:
```diff
@@ -0,0 +1,177 @@
+
+Each Mattermost installation comes with some built-in slash commands that are ready to use. These commands are available in the [latest Mattermost release](https://mattermost.com/download):
+
+.. csv-table::
+    :header: "Command", "Description", "Example"
```
Broken markdown

hanzei commented on site/content/integrate/admin-guide/admin-slash-commands/_index.md:
```diff
@@ -0,0 +1,177 @@
+
+Mattermost supports slash commands to easily integrate external applications into the server. They function similarly to [outgoing webhooks](../admin-webhooks-outgoing), except they can be used in any channel - including Private channels and Direct Messages.
+
+Messages that begin with `/` are interpreted as slash commands. The commands will send an HTTP POST request to a web service, and process a response back to Mattermost. Mattermost supports both [built-in](https://docs.mattermost.com/developer/slash-commands.html#built-in-commands) and [custom slash commands](https://docs.mattermost.com/developer/slash-commands.html#custom-slash-command).
```
The link should target mattermost.dev now

hanzei commented on site/content/integrate/admin-guide/admin-slash-commands/_index.md:
```diff
@@ -0,0 +1,177 @@
+---
```
Can we consolidate this whole page with https://developers.mattermost.com/integrate/slash-commands/?

hanzei commented on site/content/integrate/admin-guide/admin-slash-commands/_index.md:
```diff
@@ -0,0 +1,177 @@
+
+1. Open **Main Menu > Integrations > Slash Commands**. If you don't have the **Integrations** option in your Main Menu, slash commands may not be enabled on your Mattermost server or may be disabled for non-admins. Enable them from **System Console > Integrations > Integration Management** or ask your Mattermost System Admin to do so.
+2. Select **Add Slash Command** and add a name and description for the command.
+3. Set the **Command Trigger Word**. The trigger word must be unique and cannot begin with a slash or contain any spaces. It also cannot be one of the [built-in commands](https://docs.mattermost.com/help/messaging/executing-commands.html#built-in-commands).
```
Link to the table above instead?

hanzei commented on site/content/integrate/admin-guide/admin-slash-commands/_index.md:
````diff
@@ -0,0 +1,177 @@
+10. To have your application post a message back to `town-square`, it can respond to the HTTP POST request with a JSON response such as:
+
+```
+   {"response_type": "in_channel", "text": "
````
Indentation

hanzei commented on site/content/integrate/admin-guide/admin-webhooks-outgoing/_index.md:
```diff
@@ -0,0 +1,151 @@
+ - It's posted in a specified channel.
+ - The first word matches or starts with one of the defined trigger words, such as `gif`.
+
+Outgoing webhooks are supported in public channels only. If you need a trigger that works in a private channel or a direct message, consider using a [slash command](../admin-slash-commands) instead.
```
Invalid link

hanzei commented on site/content/integrate/admin-guide/admin-webhooks-outgoing/_index.md:
```diff
@@ -0,0 +1,151 @@
+
+![image](images/webhooksTable.png)
+
+11. You're all set! See [developer documentation](https://developers.mattermost.com/integrate/outgoing-webhooks) for details on what parameters are supported by outgoing webhooks. For instance, you can override the username and profile picture the messages post as, or specify a custom post type when sending a webhook message for use by [plugins](https://about.mattermost.com/default-plugins).
```
https://about.mattermost.com/default-plugins should be replace by an internal link

hanzei commented on site/content/integrate/admin-guide/admin-webhooks-outgoing/_index.md:
```diff
@@ -0,0 +1,151 @@
+2. If the text in the JSON response is longer than the allowable character limit per post, the message is split into multiple consecutive posts, each within the character limit. Servers running Mattermost Server v5.0 or later [can support posts up to 16383 characters](https://docs.mattermost.com/administration/important-upgrade-notes.html).
+3. Outgoing webhooks are supported in public channels only. If you need a trigger that works in a Private Channel or a Direct Message, consider using a [slash command](../admin-slash-commands) instead.
+4. You can restrict who can create outgoing webhooks in [System Console > Integrations > Integration Management](https://docs.mattermost.com/administration/config-settings.html#restrict-managing-integrations-to-admins).
+5. Mattermost outgoing webhooks are Slack-compatible. You can copy-and-paste code used for a Slack outgoing webhook to create Mattermost integrations. Mattermost [automatically translates the Slack's proprietary JSON response format](https://docs.mattermost.com/developer/webhooks-outgoing.html?highlight=translate%20slack%20data%20format%20mattermost#translate-slack-s-data-format-to-mattermost).
```
Bad link

hanzei commented on site/content/integrate/admin-guide/admin-webhooks-outgoing/_index.md:
```diff
@@ -0,0 +1,151 @@
+1. Webhooks are designed to easily allow you to post messages. For other actions such as channel creation, you must also use the [Mattermost APIs](https://api.mattermost.com).
+2. If the text in the JSON response is longer than the allowable character limit per post, the message is split into multiple consecutive posts, each within the character limit. Servers running Mattermost Server v5.0 or later [can support posts up to 16383 characters](https://docs.mattermost.com/administration/important-upgrade-notes.html).
+3. Outgoing webhooks are supported in public channels only. If you need a trigger that works in a Private Channel or a Direct Message, consider using a [slash command](../admin-slash-commands) instead.
+4. You can restrict who can create outgoing webhooks in [System Console > Integrations > Integration Management](https://docs.mattermost.com/administration/config-settings.html#restrict-managing-integrations-to-admins).
```
Is this still valid?

hanzei commented on site/content/integrate/admin-guide/admin-zapier-integration/_index.md:
````diff
@@ -0,0 +1,149 @@
+    [Body of issue or comment]
+```
+
+.. image:: ../images/zapier-ch1.png
````
Needs markdown image

hanzei commented on site/content/integrate/admin-guide/admin-zapier-integration/_index.md:
````diff
@@ -0,0 +1,149 @@
+
+- **Markdown:** Mattermost supports the use of [Markdown](https://docs.mattermost.com/help/messaging/formatting-text.html) in Zapier integrations. For example, use [heading markdown](https://docs.mattermost.com/help/messaging/formatting-text.html#headings) for Jira issue titles.
+- **Custom Icons:** Use different icons for different services and Zapier integrations.
+- **Hashtags:** Use hashtags to make your Zapier posts searchable. Use different hashtags for different services and Zapier integrations. For example, use the dynamic fields available in Zapier to include ticket a Jira ticket number in hashtags. This makes all conversation on a specific ticket instantly searchable by selecting the hashtag.```
````
```suggestion
- **Hashtags:** Use hashtags to make your Zapier posts searchable. Use different hashtags for different services and Zapier integrations. For example, use the dynamic fields available in Zapier to include ticket a Jira ticket number in hashtags. This makes all conversation on a specific ticket instantly searchable by selecting the hashtag.
```

hanzei commented on site/content/integrate/admin-guide/admin-zapier-integration/_index.md:
```diff
@@ -0,0 +1,149 @@
+
+If you'd like to deauthorize Zapier so it can no longer post through your connected account, go to **Account Settings > Security > OAuth 2.0 Applications**, then select **Deauthorize** on the Zapier app.
+
+![image](zapier-deauthorize.png)
```
Broken image

hanzei commented on site/content/integrate/admin-guide/admin-plugins-beta/_index.md:
```diff
@@ -0,0 +1,282 @@
+---
```
Should we consolidate with https://developers.mattermost.com/extend/plugins/?

hanzei commented on site/content/integrate/admin-guide/admin-plugins-beta/_index.md:
```diff
@@ -0,0 +1,282 @@
+
+The Plugin Marketplace is available from v5.16 and is accessed via **Main Menu > Plugin Marketplace**. More listings will be added as we add new features and plugins that our customers request.
+
+.. image:: https://user-images.githubusercontent.com/915956/66891467-1b18eb80-ef9e-11e9-9de3-37a3c5899bd8.png
```
broken image

hanzei commented on site/content/integrate/admin-guide/admin-plugins-beta/_index.md:
```diff
@@ -0,0 +1,282 @@
+
+There are two settings in **System Console > Plugin Management**:
+
+.. image:: https://user-images.githubusercontent.com/915956/66892854-94660d80-efa1-11e9-805c-a85223d43a07.png
```
broken image

hanzei reviewed (changes requested):
Thanks for consolidating all the docs :+1: 

A few high level comments:
1. Can we simplify all directory names from e.g. `admin-message-attachments` to `message-attachments`?
2. Why are all the new docs nested under `admin-guide`? Most of the content actually targets devs, not admins.
3. It's weird to see two pages talk about slash commands and webhoks, if the content is mostly the same
4. The Indentation is off by 2 spaces is most code sections

justinegeffen commented:
Thanks for the feedback, @hanzei!

Some answers to your questions:
1. *Can we simplify all directory names from e.g. admin-message-attachments to message-attachments?* - we definitely can. This naming was done because I consolidated the docs from the Admin guide and didn't want any confusion between the current docs and these either when they're being updated by me or by a community member. 
2. *Why are all the new docs nested under admin-guide? Most of the content actually targets devs, not admins.* - Same reason as above - they're from the admin guide. What I still need to do is rename the sections to: Getting Started, Developer Guide, Apps, and Integrations Admin Guide. There are different audiences for the docs although not clearly delineated. This is one way to try cater to all of them.
3. *It's weird to see two pages talk about slash commands and webhoks, if the content is mostly the same* - You're absolutely right and that's one reason I pulled this all together. There are a few options for us to think about. One is, we can fold all the content (e.g. all the webhooks content) together and make it one guide. However, the current split is one set of docs is more admin-focused (as it mentions in the intro of most of them) and the other is more dev-focused. Personally, I'd like to keep them split but also audit them and make sure there's no duplication of content. That said, I am open to suggestions and we can set up a call to talk through this with the team. :)
4. *The Indentation is off by 2 spaces is most code sections* - thank you! The rst > md conversion was a bit of a pain so there are some errors. I'll sort that out. 

1.

hanzei commented:
1. Would you mind fixing the file name and we talk about 2. and 3. 
2. Not a blocker, but I would like to talk about this point.
3. Would you mind queuing an item for discussion? Maybe for Mattercon?
4. :+1:

justinegeffen commented on site/content/integrate/admin-guide/archive/README.md:
```diff
@@ -0,0 +1 @@
+
```
I've added that for the files I transferred over and kept in `.md` format in case the new structure didn't work out. Happy to delete it when I've confirmed it's all displaying as needed.

justinegeffen commented on site/content/integrate/admin-guide/admin-bot-accounts/_index.md:
```diff
@@ -0,0 +1,185 @@
+---
+title: "Bot Accounts"
+heading: "Introduction"
```
Yep, this is copied and pasted. I'll edit them.

justinegeffen commented on site/content/integrate/admin-guide/_index.md:
```diff
@@ -0,0 +1,132 @@
+
+**Note:** The Mattermost development team is also working on an [API developer token](https://docs.google.com/document/d/1ey4eNQmwK410pNTvlnmMWTa1fqtj8MV4d9XkCumI384), which allows you to authenticate the bot account via the API token rather than retrieving a session token from a user account.
+
+## How should I automate the install and upgrade of Mattermost when included in another application?
```
Great question. I just copied all this content across and haven't done a deep review. Let's leave it in for now and I'll find a new place for it in a separate PR.

justinegeffen commented on site/content/integrate/admin-guide/admin-golang-webhook/_index.md:
```diff
@@ -0,0 +1,46 @@
+---
```
This was also a direct copy and paste from the existing content to here. Happy to remove it or fold it into something if that works better.

justinegeffen commented:
@hanzei: thank you for your extremely thorough review as usual. :) 

1. I've revised all the header/intro sections.
2. With regard to the indenting for the code, I suspect it's because I copied and pasted and lost the formatting in translation. Would you be open to helping to fix that in a new PR? 
3. I'd like to propose merging this PR - it's become a bit unwieldy to review. Then, once merged, I'll review your feedback on this PR and apply it to the published content to ensure it's all addressed. 
4. Going forward, I'm open to folding content into more defined sections or delineating the content more clearly.

LMK your thoughts. :)

hanzei reviewed (dismissed):
It seems like a lot of the issue I've pointed out in my first review like https://github.com/mattermost/mattermost-developer-documentation/pull/848#discussion_r652632615 are not fixed yet. Did you intentionally deffer them? I very hesitant to merge broken images an links.


Please also note that the LHS is broken due to the `archive` directory

![Screenshot from 2021-06-23 08-47-46](https://user-images.githubusercontent.com/16541325/123096769-b75fea80-d42f-11eb-8541-c99ee8f3706d.png)

justinegeffen commented:
Nope, not intentionally deferred, sorry! Definitely agree that merging with broken links and images isn't the goal - I haven't had a moment to review these. I'll remove the archive directory as well.

justinegeffen commented:
@hanzei, I removed the archive directory and everything is displaying better. I still can't see anything in preview - not sure why so I can't check the refs I added to the interactive dialogs page. Would you be open to checking them?

hanzei commented on site/content/integrate/admin-guide/_index.md:
```diff
@@ -0,0 +1,167 @@
+ - The ability to build plugins similar to the webapp but for React Native apps for iOS and Android.
+ - A system or architecture to combine the above plugins into one easy-to-share and easy-to-install package.
+ - A market or directory to find official and/or certified by Mattermost plugins and a process to get your plugin certified.
+ - A guide or system to enable the embedding of Mattermost into other apps as a chat service.
```
I'm hesitant to publish this as a planed item given that it's not on the roadmap

hanzei reviewed (commented):
The `ref`'s work :+1:

justinegeffen commented on site/content/integrate/admin-guide/admin-zapier-integration/_index.md:
````diff
@@ -0,0 +1,149 @@
+    [Body of issue or comment]
+```
+
+.. image:: ../images/zapier-ch1.png
````
```suggestion
![image](zapier-ch1.png)
```

justinegeffen commented on site/content/integrate/admin-guide/admin-oauth2/_index.md:
```diff
@@ -0,0 +1,80 @@
+1. Log in to your Mattermost server as the System Administrator.
+2. Go to **System Console > Integrations > Integration Management**.
+3. Set [Enable OAuth 2.0 Service Provider](https://docs.mattermost.com/administration/config-settings.html#enable-oauth-2-0-service-provider) to **True**.
+4. (Optional) If you'd like to allow external applications to post with customizable usernames and profile pictures, then set [Enable integrations to override usernames](https://docs.mattermost.com/administration/config-settings.html#enable-webhooks-and-slash-commands-to-override-usernames) and [Enable integrations to override profile picture icons](https://docs.mattermost.com/administration/conf ...
```
```suggestion
4. (Optional) If you'd like to allow external applications to post with customizable usernames and profile pictures, then set [Enable integrations to override usernames](https://docs.mattermost.com/administration/config-settings.html#enable-webhooks-and-slash-commands-to-override-usernames) and [Enable integrations to override profile picture icons](https://docs.mattermost.com/configure/configuration-settings.html#enable-integrations-to-override-profile-picture-icons) to **true**.
```

justinegeffen replied:
```suggestion
4. (Optional) If you'd like to allow external applications to post with customizable usernames and profile pictures, then set [Enable integrations to override usernames](https://docs.mattermost.com/configure/configuration-settings.html#enable-integrations-to-override-usernames) and [Enable integrations to override profile picture icons](https://docs.mattermost.com/configure/configuration-settings.html#enable-integrations-to-override-profile-picture-icons) to **True**.
```

justinegeffen commented:
@jasonblais, adding you as a reviewer to get this merged so it can be linked from the product docs.

cwarnermm reviewed (approved):
Great job on this big project!
